pkg/provider/aws/resources/api_gateway.go

```diff
@@ -85,6 +87,278 @@ type (
 	}
 )
 
+type RestApiCreateParams struct {
+	AppName string
+	Refs    []core.AnnotationKey
+	Name    string
+}
+
+func (api *RestApi) Create(dag *core.ResourceGraph, params RestApiCreateParams) error {
+
+	name := restApiSanitizer.Apply(fmt.Sprintf("%s-%s", params.AppName, params.Name))
+	if params.AppName == "" {
```

Comment:
Should this ever happen?

Reply:
yeah because in the expansion we have the api name we just pass that directly in as the name so this way we can get the correct thing. I guess we could reverse engineer the name (sounds complex but just string replacement lol) to get the app name which might be cleaner. I think my takeaway is that our resources should also be namespaced by app for ease to work with

Reply:
> we just pass that directly in as the name so this way we can get the correct thing

Not too sure what this means, can you point me to where that happens?

Reply:
its in the following pr, during the expansion for the integration https://github.com/klothoplatform/klotho/pull/624/files#diff-ab62ae487c13aa3db6ea2836ee2df671058485d34b5ee115a0fd67813ec5bc37R469